sdk/src/client/api/block_builder/transaction_builder/mod.rs

```diff
@@ -40,12 +46,133 @@ use crate::{
+                if cursor.is_none() {
+                    break;
+                }
+            }
+        }
+        let available_inputs = self
+            .get_outputs_with_metadata(&available_input_ids.into_iter().collect::<Vec<_>>())
+            .await?
+            .into_iter()
+            .map(|res| InputSigningData {
+                output: res.output,
+                output_metadata: res.metadata,
+                chain: None,
+            })
+            .collect::<Vec<_>>();
+
+        let mut mana_rewards = HashMap::new();
+
+        if let Some(burn) = &options.burn {
+            for delegation_id in burn.delegations() {
+                let output_id = self.delegation_output_id(*delegation_id).await?;
+                mana_rewards.insert(
+                    output_id,
+                    self.get_output_mana_rewards(&output_id, slot_commitment_id.slot_index())
+                        .await?
+                        .rewards,
+                );
+            }
+        }
+
+        // Check that no input got already locked
+        for output_id in &options.required_inputs {
+            let input = self.get_output(output_id).await?;
+            if input.output.can_claim_rewards(outputs.iter().find(|o| {
+                input
+                    .output
+                    .chain_id()
+                    .map(|chain_id| chain_id.or_from_output_id(output_id))
+                    == o.chain_id()
+            })) {
+                mana_rewards.insert(
+                    *output_id,
+                    self.get_output_mana_rewards(output_id, slot_commitment_id.slot_index())
+                        .await?
+                        .rewards,
+                );
+            }
+        }
+
+        let mut transaction_builder = TransactionBuilder::new(
+            available_inputs,
+            outputs,
+            addresses,
+            creation_slot,
+            slot_commitment_id,
+            protocol_parameters.clone(),
+        )
+        .with_required_inputs(options.required_inputs)
+        .with_context_inputs(options.context_inputs)
+        .with_mana_rewards(mana_rewards)
+        .with_payload(options.tagged_data_payload)
+        .with_mana_allotments(options.mana_allotments)
+        .with_remainder_address(remainder_address)
+        .with_burn(options.burn);
+
+        if let (Some(account_id), Some(reference_mana_cost)) = (options.issuer_id, reference_mana_cost) {
+            transaction_builder = transaction_builder.with_min_mana_allotment(account_id, reference_mana_cost);
+        }
+
+        if !options.allow_additional_input_selection {
+            transaction_builder = transaction_builder.disable_additional_transaction_builder();
+        }
+
+        if let Some(capabilities) = options.capabilities {
+            transaction_builder = transaction_builder.with_transaction_capabilities(capabilities)
+        }
+
+        let prepared_transaction_data = transaction_builder.select()?;
```

Comment:
Rename to build/finish?

Reply:
Absolutely my fault and I'm happy to do the change for you but we should actually go with `finish` IMO. In the rest of the codebase, build usually gives us a builder and finish gives us the final type.

Reply:
alright